hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java

```diff
@@ -1485,6 +1487,10 @@ Writable call(RPC.RpcKind rpcKind, Writable rpcRequest,
         releaseAsyncCall();
       }
       throw e;
+    } finally {
+      if (!success) {
+        connection.calls.remove(call.id);
```

Comment:
sendRpcRequest(call) will try to add a pair of <call,buf> into rpcRequestQueue. Should we remove that on failure as well?

`      rpcRequestQueue.put(Pair.of(call, buf));
`

Reply:
Thanks @xinglin for your comment. 

We don't need to remove the call from `rpcRequestQueue`, because the `rpcRequestThread` will poll it and send to the connection.

Reply:
Then the change makes sense to me. Thanks,